### PR TITLE
Add python 3.9 to CircleCI tests

### LINKED ISSUE #1263: Add CI test for Python 3.9
- With the release of Python 3.9, we should test our CI in an environment with Python 3.9
- https://www.python.org/downloads/release/python-390/

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1263 

@ParthivNaresh FYI!

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ clean:
 
 .PHONY: html
 html:
-	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html $(SPHINXOPTS)
+	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html $(SPHINXOPTS) -j 'auto'
```
Comment (angela97lin) [inline docs/Makefile:21]: Oops, accidentally removed in https://github.com/alteryx/evalml/pull/1959 so adding back in

**Review by chukarsten — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -147,7 +147,7 @@ workflows:
       - unit_tests:
           matrix:
             parameters:
-              python_version: [ "3.7" ]
+              python_version: [ "3.7", "3.9" ]
```
Comment (jeremyliweishih) [inline .circleci/config.yml:150]: is this added here because we rely on `3.8` for codecov?

Comment (angela97lin) [reply .circleci/config.yml:150]: Yup! We also don't test w/ core dependencies as true

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Pull request merged**